Updated ast.MemberExpr, equal, and added more tests

MichaelS11 opened the pull request.

MichaelS11 commented:
@mattn ,

Any thoughts on this?

mattn commented:
Please note what you want to change. AFAICS, you seems to want to change `nil == nil` and add `kindOf` ?

MichaelS11 commented:
Added toInt, updated error codes, changed so return can be used in main, added kindOf, updated member expressions and index expressions. Those are the bigger changes. The update to IsValid seemed fairly minor, not sure if it really changed much.

mattn commented on builtins/regexp/regexp.go:
```diff
@@ -8,7 +8,7 @@ import (
 
 func Import(env *vm.Env) *vm.Env {
-	m := env.NewPackage("sort")
+	m := env.NewPackage("regexp")
```
good catch. thank you

MichaelS11 replied:
Welcome :)

MichaelS11 commented:
Note that the changes are grouped into there own commit. The addition of new tests or test changes are also in there own commit. The hope was that would make it easier to review. Please let me know if there is anything else I can do to help make the changes easier to review.

mattn commented:
LGTM. Thank you. In next time for PR, please note what you want to change in description.
